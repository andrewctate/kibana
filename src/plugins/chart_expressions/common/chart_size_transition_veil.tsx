/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { useCallback, useRef, useState } from 'react';
import fastIsEqual from 'fast-deep-equal';
import type { ChartSizeSpec } from './types';

/**
 * This hook is used to show a veil over the chart while it is being resized
 * in response to a change in the container dimensions.
 *
 * It is only relevant if client dimensions are being requested based on chart configuration.
 */
export function useSizeTransitionVeil(
  chartSizeSpec: ChartSizeSpec,
  setChartSize: (d: ChartSizeSpec) => void
) {
  const [showVeil, setShowVeil] = useState(false);
  const currentChartSizeSpec = useRef<ChartSizeSpec>();

  if (!fastIsEqual(chartSizeSpec, currentChartSizeSpec.current)) {
    // If the dimensions have changed we request new dimensions from the client
    // and set off a chain of events:
    //
    // 1. we show the veil to hide step 4
    // 2. the charts library will plan a render
    // 3. the client will resize the container
    // 4. the charts library will render the chart based on the original container dimensions
    // 5. the charts library will resize the chart to the updated container dimensions
    // 6. we hide the veil

    setChartSize(chartSizeSpec);
    setShowVeil(true);
    currentChartSizeSpec.current = chartSizeSpec;
  }

  const onResize = useCallback(() => {
    setShowVeil(false);
  }, []);

  return {
    veil: (
      <div
        css={{
          height: '100%',
          width: '100%',
          backgroundColor: 'white',
          position: 'absolute',
          zIndex: 1,
          display: showVeil ? 'block' : 'none',
        }}
      />
    ),
    onResize,
  };
}
