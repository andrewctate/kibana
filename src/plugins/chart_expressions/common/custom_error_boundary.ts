/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';

interface Props {
  onError: (error: Error) => void;
}

export class CustomErrorBoundary extends React.Component<Props> {
  onError: Props['onError'];

  constructor(props: Props) {
    super(props);
    this.onError = props.onError;
  }

  componentDidCatch(error: Error) {
    this.onError(error);
  }

  render() {
    return this.props.children;
  }
}
