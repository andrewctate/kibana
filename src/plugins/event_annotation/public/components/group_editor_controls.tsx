/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import {
  EuiButton,
  EuiFieldText,
  EuiForm,
  EuiFormRow,
  EuiSelect,
  EuiSpacer,
  EuiText,
  EuiTextArea,
  useEuiTheme,
} from '@elastic/eui';
import { css } from '@emotion/react';
import type { DataView, DataViewSpec } from '@kbn/data-views-plugin/common';
import { i18n } from '@kbn/i18n';
import { FormattedMessage } from '@kbn/i18n-react';
import { SavedObjectsTaggingApiUiComponent } from '@kbn/saved-objects-tagging-oss-plugin/public';
import React, { useEffect, useMemo, useState } from 'react';
import { EventAnnotationConfig, EventAnnotationGroupConfig } from '../../common';
import { AnnotationsPanel } from './annotations_config_panel';

export const GroupEditorControls = ({
  group,
  update,
  setSelectedAnnotation,
  selectedAnnotation,
  TagSelector,
  dataViews: globalDataViews,
  createDataView,
}: {
  group: EventAnnotationGroupConfig;
  update: (group: EventAnnotationGroupConfig) => void;
  selectedAnnotation: EventAnnotationConfig | undefined;
  setSelectedAnnotation: (annotation: EventAnnotationConfig) => void;
  TagSelector: SavedObjectsTaggingApiUiComponent['SavedObjectSaveModalTagSelector'];
  dataViews: DataView[];
  createDataView: (spec: DataViewSpec) => Promise<DataView>;
}) => {
  const { euiTheme } = useEuiTheme();

  // save the spec for the life of the component since the user might change their mind after selecting another data view
  const [adHocDataView, setAdHocDataView] = useState<DataView>();

  useEffect(() => {
    if (group.dataViewSpec) {
      createDataView(group.dataViewSpec).then(setAdHocDataView);
    }
  }, [createDataView, group.dataViewSpec]);

  const dataViews = useMemo(() => {
    const items = [...globalDataViews];
    if (adHocDataView) {
      items.push(adHocDataView);
    }
    return items;
  }, [adHocDataView, globalDataViews]);

  const currentDataView = useMemo(
    () => dataViews.find((dataView) => dataView.id === group.indexPatternId) || dataViews[0],
    [dataViews, group.indexPatternId]
  );

  return !selectedAnnotation ? (
    <>
      <EuiText
        size="s"
        css={css`
          margin-bottom: ${euiTheme.size.base};
        `}
      >
        <h4>
          <FormattedMessage id="eventAnnotation.groupEditor.details" defaultMessage="Details" />
        </h4>
      </EuiText>
      <EuiForm>
        <EuiFormRow
          label={i18n.translate('eventAnnotation.groupEditor.title', {
            defaultMessage: 'Title',
          })}
        >
          <EuiFieldText
            data-test-subj="annotationGroupTitle"
            value={group.title}
            onChange={({ target: { value } }) =>
              update({
                ...group,
                title: value,
              })
            }
          />
        </EuiFormRow>
        <EuiFormRow
          label={i18n.translate('eventAnnotation.groupEditor.description', {
            defaultMessage: 'Description',
          })}
        >
          <EuiTextArea
            data-test-subj="annotationGroupDescription"
            value={group.description}
            onChange={({ target: { value } }) =>
              update({
                ...group,
                description: value,
              })
            }
          />
        </EuiFormRow>
        <EuiFormRow>
          <TagSelector
            initialSelection={group.tags}
            onTagsSelected={(tags: string[]) =>
              update({
                ...group,
                tags,
              })
            }
          />
        </EuiFormRow>
        <EuiFormRow
          label={i18n.translate('eventAnnotation.groupEditor.dataView', {
            defaultMessage: 'Data view',
          })}
        >
          <EuiSelect
            data-test-subj="annotationDataViewSelection"
            options={dataViews.map(({ id: value, title, name }) => ({
              value,
              text: name ?? title,
            }))}
            value={group.indexPatternId}
            onChange={({ target: { value } }) =>
              update({
                ...group,
                indexPatternId: value,
                dataViewSpec: value === adHocDataView?.id ? adHocDataView.toSpec(false) : undefined,
              })
            }
          />
        </EuiFormRow>
        <EuiFormRow label="Annotations">
          <div>
            {group.annotations.map((annotation) => (
              <>
                <EuiButton fullWidth onClick={() => setSelectedAnnotation(annotation)}>
                  {annotation.label}
                </EuiButton>
                <EuiSpacer size="s" />
              </>
            ))}
          </div>
        </EuiFormRow>
      </EuiForm>
    </>
  ) : (
    <AnnotationsPanel
      annotation={selectedAnnotation}
      onAnnotationChange={(changes) => setSelectedAnnotation({ ...selectedAnnotation, ...changes })}
      dataView={currentDataView}
      getDefaultRangeEnd={(rangeStart) => rangeStart}
    />
  );
};
