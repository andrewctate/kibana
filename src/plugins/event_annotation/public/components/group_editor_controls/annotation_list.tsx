/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { css } from '@emotion/react';
import { DragContext, DragDrop, ReorderProvider } from '@kbn/dom-drag-drop';
import {
  DimensionButton,
  DimensionTrigger,
  EmptyDimensionButton,
} from '@kbn/visualization-ui-components/public';
import React, { useContext, useEffect, useState } from 'react';
import { v4 as uuidv4 } from 'uuid';
import { euiThemeVars } from '@kbn/ui-theme';
import { i18n } from '@kbn/i18n';
import { createCopiedAnnotation, EventAnnotationConfig } from '../../../common';
import { getAnnotationAccessor } from '..';

export const AnnotationList = ({
  annotations,
  selectAnnotation,
  update: updateAnnotations,
}: {
  annotations: EventAnnotationConfig[];
  selectAnnotation: (annotation: EventAnnotationConfig) => void;
  update: (annotations: EventAnnotationConfig[]) => void;
}) => {
  const [newAnnotationId, setNewAnnotationId] = useState<string>(uuidv4());
  useEffect(() => {
    setNewAnnotationId(uuidv4());
  }, [annotations.length]);

  const { dragging } = useContext(DragContext);

  const reorderableGroup = annotations.map(({ id }) => ({
    id,
  }));

  return (
    <div>
      <ReorderProvider id="annotationsGroup">
        {annotations.map((annotation, index) => (
          <div
            key={index}
            css={css`
              margin-top: ${euiThemeVars.euiSizeS};
              position: relative; // this is to properly contain the absolutely-positioned drop target in DragDrop
            `}
          >
            <DragDrop
              order={[index]}
              key={annotation.id}
              value={{
                id: annotation.id,
                humanData: {
                  label: annotation.label,
                },
              }}
              dragType="move"
              dropTypes={dragging ? ['reorder'] : []}
              reorderableGroup={reorderableGroup}
              draggable
              onDrop={(dropped) => {
                const newAnnotations = [...annotations];
                const sourceIndex = newAnnotations.findIndex(({ id }) => id === dropped.id);
                const targetIndex = index;

                if (sourceIndex !== targetIndex) {
                  const temp = newAnnotations[sourceIndex];
                  newAnnotations[sourceIndex] = newAnnotations[targetIndex];
                  newAnnotations[targetIndex] = temp;
                }

                updateAnnotations(newAnnotations);
              }}
            >
              <DimensionButton
                groupLabel={i18n.translate('eventAnnotation.groupEditor.addAnnotation', {
                  defaultMessage: 'Annotations',
                })}
                onClick={() => selectAnnotation(annotation)}
                onRemoveClick={() =>
                  updateAnnotations(annotations.filter(({ id }) => id !== annotation.id))
                }
                accessorConfig={getAnnotationAccessor(annotation)}
                label={annotation.label}
              >
                <DimensionTrigger label={annotation.label} />
              </DimensionButton>
            </DragDrop>
          </div>
        ))}
      </ReorderProvider>

      <div
        css={css`
          margin-top: ${euiThemeVars.euiSizeS};
        `}
      >
        <EmptyDimensionButton
          dataTestSubj="addAnnotation"
          label="Add annotation"
          ariaLabel="Add annotation"
          onClick={() => {
            const newAnnotation = createCopiedAnnotation(newAnnotationId, new Date().toISOString());

            selectAnnotation(newAnnotation);
            updateAnnotations([...annotations, newAnnotation]);
          }}
        />
      </div>
    </div>
  );
};
