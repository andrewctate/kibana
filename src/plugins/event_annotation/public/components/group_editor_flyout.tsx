/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import {
  EuiFlyout,
  EuiFlyoutHeader,
  EuiTitle,
  EuiFlyoutBody,
  EuiFlyoutFooter,
  EuiFlexGroup,
  EuiFlexItem,
  EuiButtonEmpty,
  EuiButton,
  htmlIdGenerator,
} from '@elastic/eui';
import React, { useMemo, useState } from 'react';
import { FormattedMessage } from '@kbn/i18n-react';
import { SavedObjectsTaggingApi } from '@kbn/saved-objects-tagging-oss-plugin/public';
import { DataViewListItem } from '@kbn/data-views-plugin/common';
import { EventAnnotationConfig, EventAnnotationGroupConfig } from '../../common';
import { GroupEditorControls } from './group_editor_controls';

export const GroupEditorFlyout = ({
  group,
  updateGroup,
  onClose,
  onSave,
  savedObjectsTagging,
  dataViewListItems,
}: {
  group: EventAnnotationGroupConfig;
  updateGroup: (newGroup: EventAnnotationGroupConfig) => void;
  onClose: () => void;
  onSave: () => void;
  savedObjectsTagging: SavedObjectsTaggingApi;
  dataViewListItems: DataViewListItem[];
}) => {
  const flyoutHeadingId = useMemo(() => htmlIdGenerator()(), []);

  const [selectedAnnotation, setSelectedAnnotation] = useState<EventAnnotationConfig>();

  return (
    <EuiFlyout onClose={onClose} size={'s'}>
      <EuiFlyoutHeader hasBorder aria-labelledby={flyoutHeadingId}>
        <EuiTitle>
          <h2 id={flyoutHeadingId}>
            <FormattedMessage
              id="eventAnnotation.groupEditor.title"
              defaultMessage="Edit annotation group"
            />
          </h2>
        </EuiTitle>
      </EuiFlyoutHeader>

      <EuiFlyoutBody>
        <GroupEditorControls
          group={group}
          update={updateGroup}
          selectedAnnotation={selectedAnnotation}
          setSelectedAnnotation={setSelectedAnnotation}
          savedObjectsTagging={savedObjectsTagging}
          dataViewListItems={dataViewListItems}
        />
      </EuiFlyoutBody>

      <EuiFlyoutFooter>
        <EuiFlexGroup justifyContent="spaceBetween">
          <EuiFlexItem grow={false}>
            <EuiButtonEmpty data-test-subj="cancelGroupEdit" onClick={onClose}>
              <FormattedMessage id="eventAnnotation.edit.cancel" defaultMessage="Cancel" />
            </EuiButtonEmpty>
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            <EuiButton iconType="save" data-test-subj="saveAnnotationGroup" fill onClick={onSave}>
              <FormattedMessage
                id="eventAnnotation.edit.save"
                defaultMessage="Save annotation group"
              />
            </EuiButton>
          </EuiFlexItem>
        </EuiFlexGroup>
      </EuiFlyoutFooter>
    </EuiFlyout>
  );
};
