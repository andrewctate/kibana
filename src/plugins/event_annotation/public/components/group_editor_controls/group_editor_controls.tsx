/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { EuiFieldText, EuiForm, EuiFormRow, EuiSelect, EuiText, EuiTextArea } from '@elastic/eui';
import { css } from '@emotion/react';
import type { DataView, DataViewSpec } from '@kbn/data-views-plugin/common';
import { i18n } from '@kbn/i18n';
import { FormattedMessage } from '@kbn/i18n-react';
import { SavedObjectsTaggingApiUiComponent } from '@kbn/saved-objects-tagging-oss-plugin/public';
import { euiThemeVars } from '@kbn/ui-theme';
import { QueryInputServices } from '@kbn/visualization-ui-components/public';
import React, { useCallback, useEffect, useMemo, useState } from 'react';
import { EventAnnotationConfig, EventAnnotationGroupConfig } from '../../../common';
import { AnnotationEditorControls } from '../annotation_editor_controls';
import { AnnotationList } from './annotation_list';

export const ENABLE_INDIVIDUAL_ANNOTATION_EDITING = false;

export const GroupEditorControls = ({
  group,
  update,
  setSelectedAnnotation: _setSelectedAnnotation,
  selectedAnnotation,
  TagSelector,
  dataViews: globalDataViews,
  createDataView,
  queryInputServices,
}: {
  group: EventAnnotationGroupConfig;
  update: (group: EventAnnotationGroupConfig) => void;
  selectedAnnotation: EventAnnotationConfig | undefined;
  setSelectedAnnotation: (annotation: EventAnnotationConfig) => void;
  TagSelector: SavedObjectsTaggingApiUiComponent['SavedObjectSaveModalTagSelector'];
  dataViews: DataView[];
  createDataView: (spec: DataViewSpec) => Promise<DataView>;
  queryInputServices: QueryInputServices;
}) => {
  // save the spec for the life of the component since the user might change their mind after selecting another data view
  const [adHocDataView, setAdHocDataView] = useState<DataView>();

  useEffect(() => {
    if (group.dataViewSpec) {
      createDataView(group.dataViewSpec).then(setAdHocDataView);
    }
  }, [createDataView, group.dataViewSpec]);

  const setSelectedAnnotation = useCallback(
    (newSelection: EventAnnotationConfig) => {
      update({
        ...group,
        annotations: group.annotations.map((annotation) =>
          annotation.id === newSelection.id ? newSelection : annotation
        ),
      });
      _setSelectedAnnotation(newSelection);
    },
    [_setSelectedAnnotation, group, update]
  );

  const dataViews = useMemo(() => {
    const items = [...globalDataViews];
    if (adHocDataView) {
      items.push(adHocDataView);
    }
    return items;
  }, [adHocDataView, globalDataViews]);

  const currentDataView = useMemo(
    () => dataViews.find((dataView) => dataView.id === group.indexPatternId) || dataViews[0],
    [dataViews, group.indexPatternId]
  );

  return !selectedAnnotation ? (
    <>
      <EuiText
        size="s"
        css={css`
          margin-bottom: ${euiThemeVars.euiSize};
        `}
      >
        <h4>
          <FormattedMessage id="eventAnnotation.groupEditor.details" defaultMessage="Details" />
        </h4>
      </EuiText>
      <EuiForm>
        <EuiFormRow
          label={i18n.translate('eventAnnotation.groupEditor.title', {
            defaultMessage: 'Title',
          })}
        >
          <EuiFieldText
            data-test-subj="annotationGroupTitle"
            value={group.title}
            onChange={({ target: { value } }) =>
              update({
                ...group,
                title: value,
              })
            }
          />
        </EuiFormRow>
        <EuiFormRow
          label={i18n.translate('eventAnnotation.groupEditor.description', {
            defaultMessage: 'Description',
          })}
        >
          <EuiTextArea
            data-test-subj="annotationGroupDescription"
            value={group.description}
            onChange={({ target: { value } }) =>
              update({
                ...group,
                description: value,
              })
            }
          />
        </EuiFormRow>
        <EuiFormRow>
          <TagSelector
            initialSelection={group.tags}
            onTagsSelected={(tags: string[]) =>
              update({
                ...group,
                tags,
              })
            }
          />
        </EuiFormRow>
        <EuiFormRow
          label={i18n.translate('eventAnnotation.groupEditor.dataView', {
            defaultMessage: 'Data view',
          })}
        >
          <EuiSelect
            data-test-subj="annotationDataViewSelection"
            options={dataViews.map(({ id: value, title, name }) => ({
              value,
              text: name ?? title,
            }))}
            value={group.indexPatternId}
            onChange={({ target: { value } }) =>
              update({
                ...group,
                indexPatternId: value,
                dataViewSpec: value === adHocDataView?.id ? adHocDataView.toSpec(false) : undefined,
              })
            }
          />
        </EuiFormRow>
        {ENABLE_INDIVIDUAL_ANNOTATION_EDITING && (
          <EuiFormRow
            label={i18n.translate('eventAnnotation.groupEditor.addAnnotation', {
              defaultMessage: 'Annotations',
            })}
          >
            <AnnotationList
              annotations={group.annotations}
              selectAnnotation={setSelectedAnnotation}
              update={(newAnnotations) => update({ ...group, annotations: newAnnotations })}
            />
          </EuiFormRow>
        )}
      </EuiForm>
    </>
  ) : (
    <AnnotationEditorControls
      annotation={selectedAnnotation}
      onAnnotationChange={(changes) => setSelectedAnnotation({ ...selectedAnnotation, ...changes })}
      dataView={currentDataView}
      getDefaultRangeEnd={(rangeStart) => rangeStart}
      queryInputServices={queryInputServices}
    />
  );
};
