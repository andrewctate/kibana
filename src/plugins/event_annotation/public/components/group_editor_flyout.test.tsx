/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { EuiButton, EuiFlyout } from '@elastic/eui';
import { EventAnnotationGroupConfig } from '../../common';
import { taggingApiMock } from '@kbn/saved-objects-tagging-oss-plugin/public/api.mock';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { GroupEditorControls } from './group_editor_controls';
import { GroupEditorFlyout } from './group_editor_flyout';
import { DataView } from '@kbn/data-views-plugin/common';

const simulateButtonClick = (component: ShallowWrapper, selector: string) => {
  (component.find(selector) as ShallowWrapper<Parameters<typeof EuiButton>[0]>).prop('onClick')!(
    {} as any
  );
};

describe('group editor flyout', () => {
  const group: EventAnnotationGroupConfig = {
    annotations: [],
    description: '',
    tags: [],
    indexPatternId: 'some-id',
    title: 'My group',
    ignoreGlobalFilters: false,
  };

  const mockTaggingApi = taggingApiMock.create();

  let component: ShallowWrapper;
  let onSave: jest.Mock;
  let onClose: jest.Mock;
  let updateGroup: jest.Mock;

  beforeEach(() => {
    onSave = jest.fn();
    onClose = jest.fn();
    updateGroup = jest.fn();
    component = shallow(
      <GroupEditorFlyout
        group={group}
        onSave={onSave}
        onClose={onClose}
        updateGroup={updateGroup}
        dataViews={[
          {
            id: 'some-id',
            title: 'My Data View',
          } as DataView,
        ]}
        savedObjectsTagging={mockTaggingApi}
        createDataView={jest.fn()}
      />
    );
  });

  it('renders controls', () => {
    expect(component.find(GroupEditorControls).props()).toMatchSnapshot();
  });
  it('signals close', () => {
    component.find(EuiFlyout).prop('onClose')({} as MouseEvent);
    simulateButtonClick(component, '[data-test-subj="cancelGroupEdit"]');

    expect(onClose).toHaveBeenCalledTimes(2);
  });
  it('signals save', () => {
    simulateButtonClick(component, '[data-test-subj="saveAnnotationGroup"]');

    expect(onSave).toHaveBeenCalledTimes(1);
  });
  it('reports group updates', () => {
    const newGroup = { ...group, description: 'new description' };
    component.find(GroupEditorControls).prop('update')(newGroup);

    expect(updateGroup).toHaveBeenCalledWith(newGroup);
  });
});
