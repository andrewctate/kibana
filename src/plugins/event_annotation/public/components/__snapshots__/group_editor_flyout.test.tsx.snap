// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`group editor flyout renders controls 1`] = `
Object {
  "createDataView": [MockFunction],
  "dataViews": Array [
    Object {
      "id": "some-id",
      "title": "My Data View",
    },
  ],
  "group": Object {
    "annotations": Array [],
    "description": "",
    "ignoreGlobalFilters": false,
    "indexPatternId": "some-id",
    "tags": Array [],
    "title": "My group",
  },
  "savedObjectsTagging": Object {
    "cache": Object {
      "getState": [MockFunction],
      "getState$": [MockFunction],
    },
    "client": Object {
      "create": [MockFunction],
      "delete": [MockFunction],
      "get": [MockFunction],
      "getAll": [MockFunction],
      "update": [MockFunction],
    },
    "ui": Object {
      "components": Object {
        "SavedObjectSaveModalTagSelector": [MockFunction],
        "TagList": [MockFunction],
        "TagSelector": [MockFunction],
      },
      "convertNameToReference": [MockFunction],
      "getSearchBarFilter": [MockFunction],
      "getTableColumnDefinition": [MockFunction],
      "getTag": [MockFunction],
      "getTagIdFromName": [MockFunction],
      "getTagIdsFromReferences": [MockFunction],
      "getTagList": [MockFunction],
      "hasTagDecoration": [MockFunction],
      "parseSearchQuery": [MockFunction],
      "updateTagsReferences": [MockFunction],
    },
  },
  "selectedAnnotation": undefined,
  "setSelectedAnnotation": [Function],
  "update": [MockFunction],
}
`;
